### PR TITLE
Add OpenAPI YAML definition support.

### PR DESCRIPTION
**Pull request by bombaywalla**
For CLJ only.

Addresses #125

### CODE REVIEW

File: clj-http-lite/project.clj [hunk 1]
```diff
@@ -6,6 +6,7 @@
   :plugins [[lein-modules "0.3.11"]]
   :dependencies [[com.github.oliyh/martian :version]
                  [org.martinklepsch/clj-http-lite "0.4.3"]
+                 [clj-commons/clj-yaml "0.7.107"]
```
Comment (oliyh) [inline clj-http-lite/project.clj:9]: Why not add the clj-yaml dependency to core? Without it, core will not technically compile on its own, and that is where the clj-yaml dependency actually is.

File: core/src/martian/yaml.clj [hunk 2]
```diff
@@ -0,0 +1,32 @@
+(ns martian.yaml
```
Comment (oliyh) [inline core/src/martian/yaml.clj:1]: I wouldn't mind a test for this ns, just at a very basic level (single route)

File: clj-http-lite/project.clj (hunk 1 above)
Comment (bombaywalla) [reply clj-http-lite/project.clj:9]: Sure. Will do.

I guess I really don't know CLJS and was not sure if I could / should have a CLJ file in something that was CLJC.

File: core/src/martian/yaml.clj (hunk 2 above)
Comment (bombaywalla) [reply core/src/martian/yaml.clj:1]: Okay

**Comment by bombaywalla** [comment]
This PR is ready for review.

**Comment by oliyh** [comment]
Hi,

I was looking at the http-kit module which has a require for Cheshire, so it should still have Cheshire in the project clj rather than relying on the user also having a dependency on core (even though they realistically would)

**Comment by bombaywalla** [comment]
Sorry if I'm being dense, but...

The `project.clj` for `martian.http-kit` has a dep for `martian.core`, which has a dep for `cheshire`.
So, if a user used `martian.http-kit` (but did not explicitly include `martian.core`) they'd still get `cheshire`, right?

**Comment by oliyh** [comment]
Hi @bombaywalla 
Yes, it would still get cheshire on the classpath but transitively. If martian.http-kit only made calls to martian.core, and martian.core made all the calls to cheshire I'd be fine with that, but martian.http-kit explicitly requires cheshire and makes calls to it. To mirror this there should be an explicit dependency on cheshire in martian.http-kit.

This will avoid the scenario where some developer removes all usage of cheshire in martian.core, and removes it from the core project.clj thinking it's no longer used. This would unintentionally and unknowingly break compilation of martian.http-kit.

Many thanks

**Comment by oliyh** [comment]
Same for clj-http-lite as well. Hato and clj-http don't need it as those libraries natively convert json to clojure.

Thanks

**Comment by bombaywalla** [comment]
Okay.  Makes sense.  Added `cheshire` back to `httpkit` and `clj-http`.
Please review.

**Comment by oliyh** [comment]
Hi,

Sorry I made a half mistake in my previous comment - clj-http-lite *does* need cheshire, clj-http doesn't.

Once that's done I think this is good to go, thank you.

**Comment by bombaywalla** [comment]
Fixed.

**Comment by oliyh** [comment]
Thank you!

**Comment by oliyh** [comment]
I'll build a snapshot to clojars tomorrow

**Comment by oliyh** [comment]
Hi, this is now in `0.1.21-SNAPSHOT`

**Comment by bombaywalla** [comment]
Thanks.

**Pull request merged**